packages/client/src/sync/fetcher/trienodefetcher.ts

```diff
@@ -0,0 +1,520 @@
+      for (let i = 0; i < rangeResult.nodes.length; i++) {
+        const receivedNode = rangeResult.nodes[i]
+        const receivedHash = bytesToHex(keccak256(receivedNode) as Uint8Array)
+        if (requestedNodes.has(receivedHash)) {
+          receivedNodes.push(rangeResult.nodes[i])
+        }
+      }
+      return Object.assign([], [receivedNodes], { completed: true })
+    } catch (e) {
+      this.debug(e)
+    }
+  }
+
+  /**
+   * Process the reply for the given job.
+   * If the reply contains unexpected data, return `undefined`,
+   * this re-queues the job.
+   * @param job fetch job
+   * @param result result data
+   */
+  process(
+    job: Job<JobTask, Uint8Array[], Uint8Array>,
+    result: TrieNodesResponse
+  ): Uint8Array[] | undefined {
+    const fullResult = (job.partialResult ?? []).concat(result)
+    job.partialResult = undefined
+    if (result.completed === true) {
+      return fullResult
+    } else {
+      // Save partial result to re-request missing items.
+      job.partialResult = fullResult
+    }
+  }
+
+  /**
+   * Store fetch result. Resolves once store operation is complete.
+   * @param result fetch result
+   */
+  async store(result: Uint8Array[]): Promise<void> {
+    this.debug('At start of store phase')
+
+    try {
+      // process received node data and request unknown child nodes
+      for (const nodeData of result[0]) {
+        const node = decodeNode(nodeData as unknown as Uint8Array)
+        const nodeHash = bytesToHex(keccak256(nodeData as unknown as Uint8Array))
+        const pathString = this.requestedNodeToPath.get(nodeHash) as string
+        const [accountPath, storagePath] = pathString!.split('/')
+        const nodePath = storagePath ?? accountPath
+        const childNodes = []
+        let unknownChildNodeCount = 0
+        let hasStorageComponent = false
+
+        // get all children of received node
+        if (node instanceof BranchNode) {
+          const children = (node as BranchNode).getChildren()
+          for (const [i, embeddedNode] of children) {
+            if (embeddedNode !== null) {
+              const newStoragePath = nodePath.concat(bytesToHex(Uint8Array.from([i])))
+              const syncPath =
+                storagePath === undefined ? newStoragePath : [accountPath, newStoragePath].join('/')
+              this.debug('branch node found')
+              childNodes.unshift({
+                nodeHash: embeddedNode,
+                path: syncPath,
+              })
+            }
+          }
+        } else if (node instanceof ExtensionNode) {
+          this.debug('extension node found')
+          const stringPath = bytesToHex(pathToHexKey(nodePath, node.key(), 'hex'))
+          const syncPath =
+            storagePath === undefined ? stringPath : [accountPath, stringPath].join('/')
+          const val = {
+            nodeHash: node.value(),
+            path: syncPath,
+          }
+          childNodes.unshift(val)
```

Comment:
any significance for using unshift than push?

Reply:
No significant reason. I just changed over to using `push` for consistency.